src/main/java/seedu/taassist/model/moduleclass/UniqueModuleClassList.java

```diff
@@ -0,0 +1,118 @@
+
+/**
+ * A list of classes that enforces uniqueness between its elements and does not allow nulls.
+ * A class is considered unique by comparing using {@code ModuleClass#isSameModuleClass(ModuleClass)}. As such,
+ * adding of classes uses ModuleClass#isSameModuleClass(ModuleClass) for equality so as to ensure that the class being
+ * added is unique in terms of identity in the UniqueModuleClassList. However, the removal of a class
+ * uses ModuleClass#equals(Object) so as to ensure that the class with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see ModuleClass#isSameModuleClass(ModuleClass)
+ */
+public class UniqueModuleClassList implements Iterable<ModuleClass> {
+
+    private final ObservableList<ModuleClass> internalList = FXCollections.observableArrayList();
+    private final ObservableList<ModuleClass> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent class as the given argument.
+     */
+    public boolean contains(ModuleClass toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameModuleClass);
+    }
+
+    /**
+     * Adds a class to the list.
+     * The class must not already exist in the list.
+     */
+    public void add(ModuleClass toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateModuleClassException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Removes the equivalent class from the list.
+     * The class must exist in the list.
+     */
+    public void remove(ModuleClass toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new ModuleClassNotFoundException();
+        }
+    }
+
+    public void setModuleClasses(seedu.taassist.model.moduleclass.UniqueModuleClassList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code classes}.
+     * {@code classes} must not contain duplicate classes.
+     */
+    public void setModuleClasses(List<ModuleClass> moduleClasses) {
+        requireAllNonNull(moduleClasses);
+        if (!moduleClassesAreUnique(moduleClasses)) {
+            throw new DuplicateModuleClassException();
+        }
+
+        internalList.setAll(moduleClasses);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<ModuleClass> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    @Override
+    public Iterator<ModuleClass> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Should ordering of module classes matter? Perhaps something to consider. If it doesn't matter, perhaps we can consider using a set instead.

Reply:
This is kept as `List` for now to be consistent with the students list. This is a possible enhancement that we can make. Currently, the search for duplicates in the list itself takes `O(n^2)` time. We can probably use a `HashSet` and improve the runtime.